### PR TITLE
Bump datadog-ci to 1.16.0

### PR DESCRIPTION
**Pull request by Drarig29**
This PR bumps `datadog-ci` to [release v1.16.0](https://github.com/DataDog/datadog-ci/releases/tag/v1.16.0).

### CODE REVIEW

File: SyntheticsRunTestsTask/package.json [hunk 1]
```diff
@@ -25,7 +25,7 @@
     "node": "10.21.0"
   },
   "dependencies": {
-    "@datadog/datadog-ci": "^1.12.0",
+    "@datadog/datadog-ci": "^1.16.0",
```
Comment (ruizb) [inline SyntheticsRunTestsTask/package.json:28]: The `^` seems "dangerous" as we tend to publish breaking changes with minor versions of datadog-ci (e.g. `failOnMissingTests` flag, `parseConfigFile` that is deprecated and will probably be removed in a minor version).

Shouldn't we use `~` instead?

**Review by ruizb — APPROVED** [review]
LGTM 🐰

File: SyntheticsRunTestsTask/package.json (hunk 1 above)
Comment (Drarig29) [reply SyntheticsRunTestsTask/package.json:28]: I didn't want to change it in this PR, but yes I agree we should either pin the version or use `~`.

Comment (Drarig29) [reply SyntheticsRunTestsTask/package.json:28]: After discussion, we decided to keep the `^`.

**Pull request merged**